syz-cluster/pkg/api/http.go

```diff
@@ -32,6 +34,29 @@ func postJSON[Req any, Resp any](ctx context.Context, url string, req *Req) (*Re
 	return finishRequest[Resp](httpReq)
 }
 
+func postMultiPartFile[Resp any](ctx context.Context, url string, reader io.Reader) (*Resp, error) {
+	// TODO: this will work well only up to some size of the file. We need a pipe and a goroutine.
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("content", "content")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a form file part: %w", err)
+	}
+	_, err = io.Copy(part, reader)
```

Comment:
```go
	
	if _, err := io.Copy(part, reader); err != nil {
		return nil, err
	}
	if err := writer.Close(); err != nil {
		return nil, err
	}
```
Approximately the same amount of lines, but gives better variables locality.
And looks more straightforward to me...

Reply:
Yes, that looks better. Updated.